File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    n := len(nums)
 
    hash := make(map[int]int)

    for i := 0 ; i < n ; i++ {
        value, ok := hash[nums[i]]
        if !ok {
            hash[nums[i]] = 1
        } else {
            hash[nums[i]] = value + 1
        }
    }

    for key, count := range hash {
        if count > n / 2 {
            return key
        }
    }

    return -1
}

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const n = nums.length;
    const hash = {};
    for ( let i = 0 ; i < n ; i++ ) {
        if ( hash[nums[i]] ) {
            hash[nums[i]] += 1;
        } else {
            hash[nums[i]] = 1;
        }
    }

    for ( let key in hash ) {
        let count = hash[key];
        if ( count > n / 2 ) {
            return +key;
        }
    }
    return -1;
};